packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -70,6 +84,15 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 
 	public constructor(client: ApplicationInsights, config?: FluidAppInsightsLoggerConfig) {
 		this.baseLoggingClient = client;
+
+		for (const filter of config?.filtering.filters ?? []) {
```

Comment:
I'm wondering how necessary this is. If someone specifies an effectively empty filter, won't the effect be that we just no-op?

Reply:
If it is important that the filters be non-empty, an alternative approach would be to define multiple types, rather than sharing a single interface. Something like:

```typescript
interface CategoryFilter {
  category: TelemetryEventCategory;
}

interface NamespaceFilter {
  namespacePattern: string;
}

type TelemetryFilter = CategoryFilter | NamespaceFilter;
```

This has the added benefit of requiring that all filters you get are "valid" (non-empty).

Reply:
I've updated to this approach, thanks for pointing that out.

Reply:
No problem! Glad it worked out!